### PR TITLE
Address deno lint

### LINKED ISSUE #62: Make `deno lint` Pass
## Summary

What: If you run `deno lint`, we will most likely have a whole bunch of errors. This issue to fix these errors by correctly formatting code, and abiding by Deno's standards 

Why: It would be good to make sure the code follows either: Deno's best practices, coding standards, or general best practices. It's a Deno project so it makes sense to utilise their tools on making the code Deno-like

## Acceptance Criteria

- `deno lint` is added to the `master.yml` CI, as a new block (similar to how we use `deno fmt --check` on some repositories)
- `deno lint` produces no errors

## Resources

[Deno Lint Documentation](https://github.com/denoland/deno_lint)

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #62 

* Updated deno to v1.1.1

* Usage of `deno lint` now passes

* Running app still works

* Added pr and issue templates

* Added master workflow to test the linting/formatting and building/starting the environment

* Created a new test docker-compose and app file so we can run the environment, start the app (to ensure the app works and runs), and to exit. I don't know of any other way, but the problem i've found is:

    * We need to wait for the docker container to run it's commands before we can do anything, and even then how do we ensure the app runs (`deno run app.ts`) **and** we can actually close the containers (due to it being CI)

### CODE REVIEW

**Comment by ebebbington** [comment]
FYI it'd be best to CR file by file, because the amount of changes `deno. fmt` made, making it virtually impossible to see what I did and what deno fmt did

**Comment by ebebbington** [comment]
> FYI it'd be best to CR file by file, because the amount of changes `deno fmt` made, making it virtually impossible to see what I did and what deno fmt did

commit by commit I mean, not file by file

File: src/resources/articles_resource.ts [hunk 1]
```diff
@@ -394,7 +392,7 @@ class ArticlesResource extends BaseResource {
       case "unset":
         favorite = await ArticlesFavoritesModel.where({
           article_id: article.id,
-          user_id: currentUser.id,
+          "user_id": currentUser.id,
```
Comment (crookse) [inline src/resources/articles_resource.ts:395]: any reason `user_id` is quoted and `article_id` isnt?

File: src/models/articles_favorites_model.ts [hunk 2]
```diff
@@ -220,9 +255,28 @@ export class ArticlesFavoritesModel extends BaseModel {
...
+          ? result.article_id
+          : 0,
+        user_id: typeof result.user_id === "number" ? result.user_id : 0,
+        value: typeof result.value === "boolean" ? result.value : false,
+        id: typeof result.id === "number" ? result.id : 0,
+      };
+      articles.push(createArticlesFavoritesModelObject(entity));
     });
+    return articles;
+    // return results.map((result: any) => {
```
Comment (crookse) [inline src/models/articles_favorites_model.ts:271]: do we plan to use this commented out code?

File: src/resources/articles_resource.ts (hunk 1 above)
Comment (ebebbington) [reply src/resources/articles_resource.ts]: No reason, not sure why I did that.. changed here: https://github.com/drashland/deno-drash-realworld-example-app/pull/65/commits/57b8a93dc7ca00aeae0e362ebed48eb607e4c17b

File: src/models/articles_favorites_model.ts (hunk 2 above)
Comment (ebebbington) [reply src/models/articles_favorites_model.ts]: Not to my knowledge, must have left it in as part of figuring out which way would pass with the lint. Removed here: https://github.com/drashland/deno-drash-realworld-example-app/pull/65/commits/57b8a93dc7ca00aeae0e362ebed48eb607e4c17b

**Comment by ebebbington** [comment]
FYI pr #71 would need to be merged **after** this one as there will probably be conflicts, as i've had to change some stuff i didn't expect to

**Pull request merged**